### ISSUE TITLE
Cinnamon 2.6 issues with three monitors

### ISSUE THREAD

**Issue by syphernl** [linuxmint/cinnamon#4453]
My Lenovo T530 supports Bumblebee/Nvidia Prime which allows to use both GPU's (Nvidia + Intel) when I put it in the dock (which has two full-size DVI ports).

This requires:
- The binary Nvidia drivers rather than nouveau
- Lightdm instead of gdm
- Unity-greeter to be removed
- Kernel 4.0 (due to ACPI bug in 3.x)
- A few minor tweaks to the BIOS.

In the past I have been using Cinnamon with the two screens (laptop screen didn't work) but since I got my third screen to work it became pretty much useless since I cannot maximize any windows on my laptop screen itself. 

If I try (either via the maximize button or by "sticking" it to the top of the screen) it simply moves to the primary screen instead. It works just fine on both "secondary" screens but once I try it on the laptop screen itself its impossible to achieve. 

I have also tried XFCE (4.12) and Gnome-shell (3.14) and both work just fine. 
Cinnamon (2.6.13) simply doesn't 

This is on Ubuntu 15.04 AMD64. If any further information is required I'm more than happy to provide this if this can lead to this issue being resolved. Right now I'm using Gnome-Shell as a (hopefully) temporary replacement until Cinnamon works.

**Comment by raqua** [comment]
I am using Mint 17.2 and 2 monitors setup and have the same issue. Maximizing on laptop screen moves window to main screen. The same happens on horizontal maximize, but not on vertical maximize. The same issue happens on tilling and snapping actions.

**Comment by Vahan86** [comment]
@syphernl, @raqua, is this still an issue for you with Cinnamon 3.0.x?

**Comment by raqua** [comment]
Unfortunately I do not have any machine with Cinnamon 3 to test at the moment.

**Comment by stubobis1** [comment]
I have the same issue. I am using a W540, and two external monitors (through the dock). Maximizing / window snapping works on the two external ones, but not the laptop screen. I am using the nvidia-361 drivers and I am running mint 18 with cinnamon version 3.0.7+sarah

This is still an issue for me.

**Comment by Vahan86** [comment]
@JosephMcc, can we remove the "Close If No Response" label and mark this issue as a bug?

**Comment by raqua** [comment]
I now have machine with CInnamon 3 and yes, it is still an issue.

**Comment by Vahan86** [comment]
@syphernl, is this still an issue for you with Cinnamon 4.2?

**Comment by raqua** [comment]
I do not have installed machine with Cinnamon 4.2 and access to 3 monitors, but I have tested it with Mint 19.2 live session and this particular issue is not there. I was able to only test the maximizing, not for example the menu issue I reported here, because obviously it was live session. 
I can not however re-arrange the monitors correctly, because when I set my main screen to the middle monitor, my the one on the right will turn off. But that is probably worth separate issue and only if it is still there after proper installation and update of the system, so for now I will not create it.
